### PR TITLE
Fix & Update broadcasting.php config

### PR DESCRIPTION
**Pull request by jlsjonas**
Part of https://github.com/laravel-enso/Notifications/issues/5 (laravel-websocket support)
'localpusher' is a performance-optimizing local push which doesn't require SSL (when using laravel-websockets

Note that server-side "encrypted" should be "scheme" (see [docs](https://github.com/pusher/pusher-http-php#pusher-constructor))

Also see https://github.com/laravel-enso/Core/pull/93

### CODE REVIEW

**Comment by aocneanu** [comment]
I aligned a bit the options / env keys and from what I'm seeing the config is identical.

Doesn't that mean that you could already use the existing env keys to point to the local instance?

And second, why did you drop the `encrypted` option? Is it obsolete?

**Comment by jlsjonas** [comment]
`Host` & `Port` were introduced, which are the main changes (back-end wise)

`encrypted` was dropped as that option is not available on the backend, `scheme` is used there (see [docs](https://github.com/pusher/pusher-http-php#pusher-constructor))

On the frontend, several extra options were defined for the same reason, notably ws & http host, ws & wss port & encrypted (where it actually goes)

File: config/broadcasting.php [hunk 1]
```diff
@@ -47,10 +47,10 @@
             'secret' => env('PUSHER_APP_SECRET'),
             'app_id' => env('PUSHER_APP_ID'),
             'options' => [
-                'host' => '127.0.0.1',
-                'port' => 6001,
+                'host' => env('PUSHER_APP_HOST', '127.0.0.1'),
+                'port' => env('PUSHER_APP_PORT', 6001),
                 'cluster' => env('PUSHER_APP_CLUSTER'),
-                'scheme' => 'http',
+                'scheme' => env('PUSHER_APP_SCHEME', 'http'),
```
Comment (jlsjonas) [inline config/broadcasting.php]: 'localpusher' purposefully doesn't implement the environment variables.

The idea is that you use `localpusher` as your `BROADCAST_DRIVER` on the machine that runs the laravel-websockets instance (`php artisan websockets:serve`, via supervisor on production) and client-side still uses your _regular_ pusher configuration, hence `PUSHER_APP_XXX` would still be set to the _external_ parameters, where the _local_ (php-side) takes advantage of a direct http connection to the laravel-websockets server for reduced resource consumption (no SSL layer, it's localhost after all)

**Comment by jlsjonas** [comment]
Note that this update depends on https://github.com/laravel-enso/Core/pull/93

**Comment by aocneanu** [comment]
@jlsjonas this is how I'm seeing the file

image

My question was, don't we achieve the same thing by using only one config and only update the ENV vars?

**Comment by jlsjonas** [comment]
> My question was, don't we achieve the same thing by using only one config and only update the ENV vars?

Please see https://github.com/laravel-enso/Enso/pull/190#discussion_r254205988
`localpusher` is used by the server while, SIMULTANEOUSLY `pusher` is being used by client-side & and/or non-local servers, in a load-balanced environment

the `localpusher part is optional in principle, but very useful for a production environment (less overhead when talking from laravel to laravel-websockets)

**Comment by aocneanu** [comment]
I don't see why we should replace `encrypted` with `scheme` in `broadcasting.php`, then to "translate" it into `secure` in a ternary (`'secure' => (config('broadcasting.connections.pusher.options.scheme') === 'https')` and in the end to pass it as a value for `Echo`'s `encrypted` option (`encrypted: this.meta.pusher.secure`)
(`encrypted` being the attribute that comes by default with Laravel, in that config file).

Regarding the `localPusher` configuration and the architecture that you are thinking about, I understand the arguments but I'm not sure that I would do that myself, at least not in most of the apps. Not being something that Enso "needs" in order to use Laravel Websockets, we will leave the `localPusher` configuration aside and let the user decide if he needs it or not.

**Comment by jlsjonas** [comment]
>I don't see why we should replace `encrypted` with `scheme` in `broadcasting.php`

As mentioned before, `encrypted` is an invalid argument for `broadcasting.php`'s config, `scheme` is **required** if you want to support http (in the back-end, at least; as it presumably defaults to https)

👍 on leaving the `localPusher` config out; it benefits only show on bigger apps which is an optimization that can be manually added anyway

**Pull request merged**